File: tests/omp/lastprivate.cpp
void lastpriv (int n, float *a, float *b)
{
  int i; 
  {
    for (i=0; i<n-1; i++)
      a[i] = b[i] + b[i+1];
  }

  a[i]=b[i];      /* i == n-1 here */
}

File: tests/omp/lastprivate.ml
open Optitrust
open Target


(* This should work fine after fixing the issue with cFor without declaration *)
let _ = Run.script_cpp (fun _ ->

  !! Omp.parallel [] [tFirst; cFunDef "lastpriv"; dBody];
  !! Omp.parallel_for [LastPrivate ["a"]] [tBefore; cFor "i"];
)
